x-pack/plugins/cloud_security_posture/public/pages/compliance_dashboard/compliance_dashboard.tsx

```diff
@@ -190,42 +195,108 @@ const getDefaultTab = (
   const kspmTotalFindings = kspmStats?.stats.totalFindings;
   const installedPolicyTemplatesCspm = pluginStatus?.cspm?.status;
   const installedPolicyTemplatesKspm = pluginStatus?.kspm?.status;
-  let preferredDashboard = CSPM_POLICY_TEMPLATE;
+  let preferredDashboard = POSTURE_TYPE_CSPM;
 
   // cspm has findings
   if (!!cspmTotalFindings) {
-    preferredDashboard = CSPM_POLICY_TEMPLATE;
+    preferredDashboard = POSTURE_TYPE_CSPM;
   }
   // kspm has findings
   else if (!!kspmTotalFindings) {
-    preferredDashboard = KSPM_POLICY_TEMPLATE;
+    preferredDashboard = POSTURE_TYPE_KSPM;
   }
   // cspm is installed
   else if (
     installedPolicyTemplatesCspm !== 'unprivileged' &&
     installedPolicyTemplatesCspm !== 'not-installed'
   ) {
-    preferredDashboard = CSPM_POLICY_TEMPLATE;
+    preferredDashboard = POSTURE_TYPE_CSPM;
   }
   // kspm is installed
   else if (
     installedPolicyTemplatesKspm !== 'unprivileged' &&
     installedPolicyTemplatesKspm !== 'not-installed'
   ) {
-    preferredDashboard = KSPM_POLICY_TEMPLATE;
+    preferredDashboard = POSTURE_TYPE_KSPM;
   }
 
   return preferredDashboard;
 };
 
-export const ComplianceDashboard = () => {
-  const [selectedTab, setSelectedTab] = useState(CSPM_POLICY_TEMPLATE);
-  const { data: getSetupStatus } = useCspSetupStatusApi();
+const determineDashboardDataRefetchInterval = (data: ComplianceDashboardData | undefined) => {
+  if (data?.stats.totalFindings === 0) {
```

Comment:
the main change is the usage of `refetchInterval` option for `useQuery` with some logic when to stop prefetching (when the data appears in the index)

Reply:
Also, maybe we should keep an interval even after we have data, like once every minute or so, @kfirpeled thoughts?

Reply:
@JordanSh I totally agree that refactorings that are not related to the fix should go separately to avoid bloated PRs and new bugs introduced by accident. The thing is that in case moving a new content `TabContent` out of `ComplianceDashboard` is also related to the fix in my opinion as I introduced the re-fetching logic for the dashboard data and it's easier to manage when it's encapsulated in the separate component, and not added to already quite complex `ComplianceDashboard` component. It's possible to add the fix without moving the tab content logic out, bu it would probably require more guards to avoid tab selection being affected by re-fetching. Plus some more guards might be needed to avoid re-fetching when it's not necessary (eg. when the plugin is not initialized). With `TabContent` component having the refetching logic it's guaranteed that we don't need to care about preventing it as it only happens when the component mounts.
I will spend some more time today and try applying the fix without moving the tab logic outside. Let's see what comes out it

Reply:
np, so lets not waste time on that. i tested it a bit and everything seemed to work the same.

Reply:
@JordanSh I will do a round of testing too to make sure I didn't break anything on the way.